### PR TITLE
Use structured logging on Fuchsia

### PR DESCRIPTION
**Pull request by mbrase**
This change migrates off of the old fuchsia logging apis to use the
structured logging apis. The initial FIDL connection is made during
global initialization (before main()) and the initial minimum log level
is queried from the system. Later on, once the main loop is initialized,
we setup an async task to listen for additional log interest changes
from the system. The advantage of doing this on the main loop is that we
avoid spawning an additional background thread in the process (the
legacy logging apis use the background thread approach).

One added benefit of this change is it reduces the size of the
dart/flutter runner far packages by about 250kb in release mode, because
libsyslog.so and libbackend_fuchsia_globals.so are no longer needed.

flutter/flutter#141924

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: fml/platform/fuchsia/log_state.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  std::shared_ptr<const std::vector<std::string>> tags() const {
+    return std::atomic_load(&tags_);
+  }
+
+  // Take ownership of the log sink channel (e.g. for LogInterestListener).
+  // This is thread-safe.
+  fidl::ClientEnd<::fuchsia_logger::LogSink> TakeClientEnd();
+
+  // Updates the default tags.
+  // This is thred-safe.
```
Comment (jrwang) [inline fml/platform/fuchsia/log_state.h:40]:
```suggestion
  // This is thread-safe.
```

File: fml/platform/fuchsia/log_state.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  void SetTags(const std::initializer_list<std::string>& tags);
+
+  // Get the default instance of LogState
```
Comment (jrwang) [inline fml/platform/fuchsia/log_state.h:43]:
```suggestion
  // Get the default instance of LogState.
```

**Review by jrwang — APPROVED** [review]
LGTM

**Comment by mbrase** [comment]
This is ready to merge, but I'm holding off until b/322387862 is resolved.

**Pull request merged**